StreamChatClient_v3/APIClient/Endpoints/ChannelEndpoints.swift

```diff
@@ -10,9 +10,9 @@ extension Endpoint {
         -> Endpoint<ChannelListEndpointPayload<ExtraData>> {
         .init(path: "channels",
               method: .get,
-              queryItems: [],
-              jsonQueryItems: ["payload": query],
-              body: nil)
+              queryItems: nil,
+              requiresConnectionId: query.options.intersection([.presence, .state, .watch]).isEmpty == false,
```

Comment:
so this `== false` is our style now? I think we should reach a common decision
cc @buh

Reply:
@b-onc  For here `isEmpty` is pretty clear.

I use `== true/false` in cases where I need to compare optional booleans, so it's easy to avoid unwrapping:
```
var isGood: Bool?

if isGood == true {
    // It's good 😌
}
```

Reply:
@buh good point! I don't think we need the verbosity except a case like you've shown.

Reply:
Here it's about the visual difference between:
```
let needsConnectionId = query.options.intersection([.presence, .state, .watch]).isEmpty == false

// vs

let needsConnectionId = !query.options.intersection([.presence, .state, .watch]).isEmpty
```
If you noticed the `!` in the beginning of the second line, you have much better eye sight than I have.

Reply:
@VojtaStavik I think you need bigger fonts / scaling for your screen :) Using FiraCode / JetBrains Mono also helps.

Reply:
Like seriously? You really think the first version is not less error-prone?

Reply:
Yep, this is a very low baseline for errors.

Reply:
OK, I solved it much better in fa17d4c10ca1c45b97c11ef504428cc9b0ceea57 and 8e5cd6a6d34803e88aba633f23f379804e010161